aztec/src/main/kotlin/org/wordpress/aztec/spans/AztecMediaSpan.kt

```diff
@@ -98,26 +98,26 @@ class AztecMediaSpan(context: Context, drawable: Drawable?, var attributes: Attr
         val sb = StringBuilder()
         sb.append("<")
         sb.append(TAG)
+        sb.append(' ')
 
-        attributes?.let {
-            for (i in 0..attributes!!.length-1) {
-                sb.append(' ')
-                sb.append(attributes!!.getLocalName(i))
-                sb.append("=\"")
-                sb.append(attributes!!.getValue(i))
-                sb.append("\"")
-            }
-        }
+        attributes.removeAttribute("aztec_id")
 
+        sb.append(attributes)
         sb.append("/>")
+
         return sb.toString()
     }
 
     fun onClick(view: View) {
+        // add a unique ID that can identify this image span
```

Comment:
> Upon click, client's mediaTapped is called. The client can perform the "if attribute missing, add a random value to it" operation in that call since the element's 

Sorry, I should have checked. You're right, the client can add it there. But that's a specific case when you have access to the `attributes` property of the span. What if you want to update an element without tapping on it? I know this is a hypothetical question but right now you wouldn't be able to do that. This could be solved if the library guaranteed the presence of an "Aztec id" for every element. What do you think>